### PR TITLE
Assignment 1

### PR DESCRIPTION
**Pull request by shivamk2406**

### CODE REVIEW

File: item-details/enum/item.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package enum
```
Comment (10yogi) [inline item-details/enum/item.go:1]: remove -detail from pack name

File: item-details/item.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+		totalCost += 0.125 * item.ItemPrice * float64(item.ItemQuantity)
+	case enum.Manufactured:
+		totalCost += 0.125 * item.ItemPrice * float64(item.ItemQuantity)
+		totalCost += 0.02 * (item.ItemPrice + 0.125*item.ItemPrice)
+	case enum.Imported:
+		totalCost = item.applySurcharge(totalCost)
+	}
+	return totalCost
+}
+
```
Comment (10yogi) [inline item-details/item.go:30]: refactor: remove input params

File: item-details/item.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+func (item Item) applySurcharge(totalCost float64) float64 {
+	totalCost += +0.1 * item.ItemPrice
+	if totalCost <= 100 {
+		totalCost += 5
+	} else if totalCost > 100 && totalCost <= 200 {
+		totalCost += 10
+	} else {
+		totalCost += 0.05 * totalCost
+	}
+	return totalCost
```
Comment (10yogi) [inline item-details/item.go:40]: define constants.
setup IDE linter.

File: item-details/item_test.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	CalculateTaxTest{i: Item{ItemName: "Pen", ItemPrice: 45, ItemQuantity: 5, ItemType: enum.Raw}, expected: 253.120000},
+	CalculateTaxTest{i: Item{ItemName: "Copy", ItemPrice: 75, ItemQuantity: 15, ItemType: enum.Manufactured}, expected: 1267.310000},
+	CalculateTaxTest{i: Item{ItemName: "Eraser", ItemPrice: 85, ItemQuantity: 16, ItemType: enum.Raw}, expected: 1530.000000},
+	CalculateTaxTest{i: Item{ItemName: "Tool", ItemPrice: 105, ItemQuantity: 18, ItemType: enum.Imported}, expected: 253.120000},
+	CalculateTaxTest{i: Item{ItemName: "Book", ItemPrice: 85.4, ItemQuantity: 29, ItemType: enum.Imported}, expected: 2609.390000},
+}
+
+func TestCalculateTax(t *testing.T) {
+	for _, test := range TaxTests {
+		output := test.i.CalculateTax(test.i.ItemPrice * float64(test.i.ItemQuantity))
```
Comment (10yogi) [inline item-details/item_test.go:25]: output??

File: item-details/item_test.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import (
+	"math"
+	"testing"
+
+	"github.com/shivamk2406/GO-Assignments/item-details/enum"
+)
+
+type CalculateTaxTest struct {
+	i        Item
+	expected float64
```
Comment (10yogi) [inline item-details/item_test.go:12]: add err component.

File: item-details/enum/item.go (hunk 1 above)
Comment (shivamk2406) [reply item-details/enum/item.go]: Done

File: item-details/item.go (hunk 2 above)
Comment (shivamk2406) [reply item-details/item.go]: Done

File: item-details/item.go (hunk 3 above)
Comment (shivamk2406) [reply item-details/item.go]: done

File: item-details/item.go [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+}
+
+func checkNegativeValue(value interface{}) error {
+	err := fmt.Errorf("%v", "negative value")
+
+	switch data := value.(type) {
+	case int:
+		if data < 0 {
+			return err
```
[the remark this answers is not in the document]
Comment (shivamk2406) [reply item-details/item.go]: Done

File: item-details/item_test.go (hunk 4 above)
Comment (shivamk2406) [reply item-details/item_test.go]: Done

File: item-details/item_test.go [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+func TestCalculateTax(t *testing.T) {
+	for _, test := range TaxTests {
+		output := test.i.CalculateTax(test.i.ItemPrice * float64(test.i.ItemQuantity))
+		output = math.Floor(output*100) / 100
+		expected := math.Floor(test.expected*100) / 100
+		if output != expected {
+			t.Errorf("Output %f not equal to expected %f", output, expected)
+		}
```
[the remark this answers is not in the document]
Comment (shivamk2406) [reply item-details/item_test.go]: Done

File: item-details/item_test.go (hunk 5 above)
Comment (shivamk2406) [reply item-details/item_test.go]: Done

File: view/view.go [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+	return nil
+
+}
+
+func getItem() (name string, price float64, quantity int, itemType string, err error) {
+
+	fmt.Println("Enter Item Name")
+	_, err = fmt.Scanf("%s", &name)
+	if err != nil {
+		err = errors.Wrap(err, "item name scanning failed")
```
[the remark this answers is not in the document]
Comment (shivamk2406) [reply view/view.go]: Done

**Pull request closed without merging**